### ISSUE TITLE
Consider API changes for the next release

### ISSUE THREAD

**Issue by kanderso-nrel** [NREL/rdtools#126]
Cutting a major release is the time to make breaking changes to the API so we should consider what (if any) we should make.  One that comes to mind is renaming the `soiling.srr_analysis` class to use CamelCase.

**Comment by kanderso-nrel** [comment]
Here's the list I've been keeping:
- Consider changing the default behavior of `normalization.irradiance_rescale` with regards to the `method` parameter (https://github.com/NREL/rdtools/pull/152#issuecomment-597772042)
- `normalization.energy_from_power` behaves differently between dev and master (https://github.com/NREL/rdtools/issues/141)
- Remove the low power cutoff in `filtering.clipping_filter` (#144)
- CamelCase `soiling.srr_analysis`.
- SRR #129 
- Consider combining all filters into one function? https://github.com/NREL/rdtools/pull/139#issuecomment-595977955

**Comment by mdeceglie** [comment]
I'd like to add to the list:
Make sure we can support the use of rainfall events as cleaning events in SRR while maintaining backward compatibility (and thus avoiding a major release).

**Comment by kanderso-nrel** [comment]
@mdeceglie Not sure I follow -- can you elaborate on what precipitation cleaning functionality we might add in the future beyond what is already implemented?

**Comment by mdeceglie** [comment]
Sorry for the vagueness. The new functionality would allow all precipitation events greater than some specified threshold to be counted as cleaning events.  

To be consistent with the way `precip_clean_only` works, we would do it without modifying the `soiling_interval_summary` dataframe. `precip_clean_only` currently works by only allowing soiling recoveries on cleaning events that coincided with precipitation thresholds, but the slopes are still found independently for the sub-parts of the interval separated by detected cleaning events that did not coincide with rain.

But thinking this through it may be overall more consistent to handle both cases earlier in the algorithm, before slopes are fit for each interval.

**Comment by matt14muller** [comment]
I gave Kevin the few lines of code I used to combine detected shifts and rain for using as intervals.
What we talked about is inserting another option for the user besided precip_clean_only....detected_clean_and_precip or something like that and the user could select this option.  The Theil-sen fits would then be calculated on these intervals (this is what I did in my version) for the soiling map.
